src/account/operations/helpers/time.rs

```diff
@@ -16,18 +19,29 @@ pub(crate) fn can_output_be_unlocked_now(
             return false;
         }
 
-        let output_address = unlock_conditions
-            .address()
-            .expect("output needs to have an address unlock condition")
-            .address();
+        if let Some(address_unlock_condition) = unlock_conditions.address() {
+            let address = address_unlock_condition.address();
 
-        let unlock_address = unlock_conditions.locked_address(output_address, current_time);
-        // The address that can unlock the output needs to belong to the account
-        if !account_addresses.iter().any(|a| a.address.inner == *unlock_address) {
-            return false;
-        };
+            let unlock_address = unlock_conditions.locked_address(address, current_time);
+            // The address that can unlock the output needs to belong to the account
+            if !account_addresses.iter().any(|a| a.address.inner == *unlock_address)
```

Comment:
contains as well :trollface: ?

Reply:
Doesn't work because we check for .address.inner, no?

Reply:
Right, overlooked it